control-base/src/main/java/fi/nls/oskari/control/layer/GetWFSLayerFieldsHandler.java

```diff
@@ -0,0 +1,68 @@
+package fi.nls.oskari.control.layer;
+
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.control.RestActionHandler;
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.map.layer.OskariLayerService;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.service.ServiceException;
+import fi.nls.oskari.service.ServiceRuntimeException;
+import fi.nls.oskari.util.ConversionHelper;
+import fi.nls.oskari.util.ResponseHelper;
+import fi.nls.oskari.util.WFSGetLayerFields;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * An action route that returns fields information for WFS layers
+ */
+@OskariActionRoute("GetWFSLayerFields")
+public class GetWFSLayerFieldsHandler extends RestActionHandler {
+    private static final String PARAM_LAYER_ID = "layer_id";
+    private OskariLayerService layerService;
+
+    @Override
+    public void init() {
+        try {
+            layerService = OskariComponentManager.getComponentOfType(OskariLayerService.class);
+        } catch (Exception e) {
+            throw new ServiceRuntimeException("Exception occurred while initializing map layer service", e);
+        }
+    }
+
+    @Override
+    public void handleAction(ActionParameters params) throws ActionException {
+        final int layerId = ConversionHelper.getInt(params.getRequiredParam(PARAM_LAYER_ID), -1);
+        final OskariLayer layer = getLayer(layerId);
+        final JSONObject response = getLayerFields(layer);
+        ResponseHelper.writeResponse(params, response);
+    }
+
+    private JSONObject getLayerFields(OskariLayer layer) throws ActionException {
+        try {
+            JSONObject fields = WFSGetLayerFields.getLayerFields(layer);
+            JSONObject locale = getFieldsLocale(layer);
+            fields.putOpt("locale", locale);
+            return fields;
+        } catch (ServiceException ex) {
+            throw new ActionException("Error getting layer fields", ex);
+        } catch (JSONException ex) {
+            throw new ActionException("Invalid attribute locale", ex);
+        }
+    }
+
+    private JSONObject getFieldsLocale(OskariLayer layer) {
+        JSONObject data = layer.getAttributes().optJSONObject("data");
+        return data != null ? data.optJSONObject("locale") : null;
+    }
+
+    private OskariLayer getLayer(int layerId) throws ActionException {
+        final OskariLayer layer = layerService.find(layerId);
```

Comment:
Run this through the PermissionHelper as in https://github.com/oskariorg/oskari-server/blob/master/control-base/src/main/java/fi/nls/oskari/control/layer/GetLayerTileHandler.java#L67

The helper will throw a ActionDeniedException if the user doesn't have permission to view the layer -> they shouldn't get to see the attributes/configurations we have on that layer either.